drivers/watchdog/Kconfig.gd32

```
@@ -0,0 +1,41 @@
+# Copyright (c) 2021 TOKITA Hiroshi <tokita.hiroshi@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_GIGADEVICE_GD32_FWDGT := gd,gd32-free-watchdog
+DT_COMPAT_GIGADEVICE_GD32_WWDGT := gd,gd32-window-watchdog
+
+config FWDGT_GD32
+	bool "GD32 Free watchdog timer (FWDGT) driver"
+	default $(dt_compat_enabled,$(DT_COMPAT_GIGADEVICE_GD32_FWDGT))
+	select USE_GD32_FWDGT
+	select USE_GD32_DBG
+	depends on (SOC_FAMILY_GD32 || SOC_SERIES_GD32VF103)
+	help
+	  Enable the Free watchdog timer (FWDGT) driver for GD32 SoCs.
+
+config FWDGT_INITIAL_TIMEOUT
+	int "Value for FWDGT timeout in ms"
+	depends on FWDGT_GD32
+	default 1000
+	help
+	  Set timeout value for FWDGT in milliseconds.
+	  Not start FWDGT if set less than 0 value to this parameter.
+	  (same behavior as CONFIG_WDT_DISABLE_AT_BOOT set.)
+	  The minimum value is 1 ms.
+	  The following equation gives the maximum value.
+
+	    prescaler * reload / peripheral_freqency.
+
+	  Where, the prescaler = 256 and the reload = 4096.
+	  Each is the maximum value that can take.
+	  peripheral_frequency depends on SoC types.
+	  For example, This is 40000 for GD32VF103 and 32000 for GD32F4xx.
```

I suggest to limit the range via Kconfig and not allow negative values where the FWDGT will not start. If the watchdog should not start during boot, the appropriate Kconfig should be used instead.
```suggestion
	range 1 40000
	default 1000
	help
	  Set timeout value for FWDGT in milliseconds.
	  The following equation gives the maximum value:

	    prescaler * reload / peripheral_freqency.

	  where the maximum prescaler = 256 and the maximum reload = 4096.
	  The peripheral_frequency depends on the SoC type.
	  For example, this is 40000 for GD32VF103 and 32000 for GD32F4xx.
```

---

FWDGT_GD32_INITIAL_TIMEOUT range changes in conjunction with IRC40K_VALUE.
GigaDevice's firmware design allows redefining IRC40K_VALUE by the user.

https://github.com/zephyrproject-rtos/hal_gigadevice/blob/242a7f4be7a3136606c7a65169775f7ef85ad444/gd32vf103/riscv/include/gd32vf103.h#L87

The maximum value of FWDGT_GD32_INITIAL_TIMEOUT defined as
   (256 * 4095 * 1000) / IRC40K_VALUE
But device-tree can use only numeric literals.
We can't define the range with that calculation in the device tree.

From the reason above, Not define the value range in the device tree and validating the FWDGT_GD32_INITIAL_TIMEOUT value range by preprocessor macros in a compilation.